### DISCUSSION TITLE
Is the following expression a valid check for `successCondition` in the ResourceTemplate

### DISCUSSION THREAD

**Discussion by nitishm** [argoproj/argo-workflows#5776] [Q&A] [answered]
https://argoproj.github.io/argo-workflows/fields/#fields_39

```go
successCondition: func() string {
   res := successCheck() // Do some checks
  if  res {
    return "true"
  }
  return "false"
}()
```

**Comment by terrytangyuan** [comment, marked as answer]
The current implementation uses https://pkg.go.dev/k8s.io/apimachinery/pkg/labels#Parse to parse the success and failure conditions. You can read more about the syntax in the documentation.

**Reply by nitishm** [reply]
Tried reading the docs but I still don't understand how it knows to parse the status field of the k8s resource. I understand labels in the context of the metadata but am having a hard time figuring out how this works.

**Reply by terrytangyuan** [reply]
It’s using https://github.com/tidwall/gjson to get the field from a k8s resource object. Related code: 
* https://github.com/argoproj/argo-workflows/blob/bdd44c723a324d1c20bcc97f53022b586bfb8348/workflow/executor/resource.go#L235
*https://github.com/argoproj/argo-workflows/blob/bdd44c723a324d1c20bcc97f53022b586bfb8348/workflow/executor/resource.go#L129
